conftest.py

```diff
@@ -615,3 +616,15 @@ def custom_alpine_image(request, session_api, zone):
         raise RuntimeError(f"Wrong SHA-256 for {url}: {sha256}")
 
     yield image
+
+
+@pytest.fixture(scope='session')
+def raw_options_call():
```

Comment:
Instead of surfacing the urllib handler, I would return a function that returns the headers for a given URL:

```python
@pytest.fixture(scope='session')
def request_headers(url):
     ....
```

Then, the tests can just use `request_headers(API_URL)` or so. In my opinion the urllib handler is an implementation detail which could change (however unlikely).

Additionally I would prefer if all tests passed the URL they are requesting explicitly. Otherwise, as a reader of the test_* files, I am left to wonder what `raw_options_call.getheaders()` actually requests.

Reply:
I removed the fixture and added a function to the test module to make the OPTIONS request. It still returns the urllib handler because the `getheaders` method is only available there.